metadata_service/adapter/datastore.py

```diff
@@ -21,18 +24,44 @@ def get_datastore_versions() -> dict:
         return json.load(f)
 
 
-def get_metadata_all(version: Version) -> str:
+def get_from_cache(file_version: str) -> dict | None:
+    if file_version == NEWEST_CACHE[0]:
+        return NEWEST_CACHE[1]
+    if file_version == LAST_REQUEST_CACHE[0]:
+        return LAST_REQUEST_CACHE[1]
+    return None
+
+
+def update_cache(file_version: str, metadata_all):
```

Comment:
can there be some problems when updating this from mulitple requests at the same time?